### PR TITLE
feat: add `Builder::chunk` method stub and test

### PR DESCRIPTION
**Pull request by amooij**
- [x] Added or updated tests
- [x] Documented user facing changes

Adds type hints for the chunk method on the Eloquent Builder.

More code checked

### CODE REVIEW

**Comment by szepeviktor** [comment]
Thank you @amooij !!

File: stubs/EloquentBuilder.stub [hunk 1]
```diff
@@ -513,6 +513,15 @@ class Builder
      * @return static
      */
     public function whereRelation($relation, $column, $operator = null, $value = null);
+
+    /**
+     * Gets the result of a query in chunks.
+     *
+     * @param  int  $count
+     * @param  callable(\Illuminate\Database\Eloquent\Collection<TModelClass>, int): bool $callback
```
Comment (canvural) [inline stubs/EloquentBuilder.stub:521]: `\Illuminate\Database\Eloquent\Collection` needs 2 generic types. I think the key here can be `array-key`

File: tests/Type/data/eloquent-builder.php [hunk 2]
```diff
@@ -432,6 +432,11 @@ function testPaginateItems()
     assertType('array<App\User>', User::query()->paginate()->items());
 }
 
+function testChunkOnEloquentBuilder()
+{
+    User::chunk(1000, fn ($collection) => assertType('Illuminate\Database\Eloquent\Collection<App\User>'));
```
Comment (canvural) [inline tests/Type/data/eloquent-builder.php:437]: After the above change, this needs to be updated also for the 2 generic types.

**Review by canvural — CHANGES_REQUESTED** [review]

File: stubs/EloquentBuilder.stub (hunk 1 above)
Comment (amooij) [reply stubs/EloquentBuilder.stub]: Done!

File: tests/Type/data/eloquent-builder.php (hunk 2 above)
Comment (amooij) [reply tests/Type/data/eloquent-builder.php]: And done!

File: tests/Type/data/eloquent-builder.php [hunk 3]
```diff
@@ -432,6 +432,11 @@ function testPaginateItems()
     assertType('array<App\User>', User::query()->paginate()->items());
 }
 
+function testChunkOnEloquentBuilder()
+{
+    User::chunk(1000, fn ($collection) => assertType('Illuminate\Database\Eloquent\Collection<int, App\User>'));
```
Comment (szepeviktor) [inline tests/Type/data/eloquent-builder.php:437]: I'm not a developer but after reading the docblock
```suggestion
    assertType('Illuminate\Database\Eloquent\Collection<int, App\User>', User::chunk(1000, fn ($collection, $someInteger) => $someBoolean));
```

Comment (szepeviktor) [inline tests/Type/data/eloquent-builder.php:437]: Is this the problem?
```suggestion
    User::chunk(1000, fn ($collection) => assertType('Illuminate\Database\Eloquent\Collection<int, App\User>', $collection));
```

**Comment by szepeviktor** [comment]
Thank you.
Now it is okay.
Do not care about that merge commit and e2e tests.

**Review by szepeviktor — APPROVED** [review]

**Comment by canvural** [comment]
I changed the return type of the callback to `mixed` Callback doesn't need to return anything, it's just if it returns false, the internal loop will stop. So `mixed` makes more sense.

Otherwise, it's good. Thank you!

**Comment by amooij** [comment]
Thanks, can you explain something to me? All the unit tests fail on my machine on master. Is this a temporary issue or something in my setup? I'm running php 8.1 on MacOS with a regular composer.

**Comment by canvural** [comment]
Can you show the errors?

**Pull request merged**